In [1]:
from choice_learn.datasets import load_expedia
import numpy as np


In [2]:
df = load_expedia(as_frame=False, preprocessing="rumnet")


c:\Users\danil\anaconda3\envs\vd200\Lib\site-packages\choice_learn\datasets\expedia.py:168: FutureWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.
  choices = expedia_df.groupby("srch_id").apply(lambda x: x.booking_bool.argmax())
c:\Users\danil\anaconda3\envs\vd200\Lib\site-packages\choice_learn\datasets\expedia.py:260: FutureWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.
  .apply(lambda x: x[contexts_items_features_names[1:-1]].to_nump

## Train test split

In [3]:
N = len(df)
rng = np.random.default_rng(42)
idx = rng.permutation(N)

n_train = int(0.8*N)
train_idx = idx[:n_train]
test_idx = idx[n_train: ]

#train and test data set
train_ds = df[train_idx]
test_df = df[test_idx]

In [7]:
train_ds.summary()

%=====================================================================%
%%% Summary of the dataset:
%=====================================================================%
Number of items: 39
Number of choices: 318094
%=====================================================================%
 Shared Features by Choice:
 13 shared features
 with names: (['srch_length_of_stay', 'srch_adults_count', 'srch_children_count', 'srch_room_count', 'srch_saturday_night_bool', 'booking_window', 'random_bool', 'day_of_week', 'month', 'hour'], ['site_id', 'visitor_location_country_id', 'srch_destination_id'])


 Items Features by Choice:
11 items features 
 with names: (['prop_starrating', 'prop_review_score', 'prop_brand_bool', 'prop_location_score1', 'prop_location_score2', 'prop_log_historical_price', 'position', 'promotion_flag', 'orig_destination_distance', 'log_price'], ['prop_country_id'])
%=====================================================================%


''

The dataset contains 318,094 choice situations and 39 items (hotels) per choice set. It includes 13 shared (context) features describing the search characteristics and 11 item-specific features describing the hotel attributes.